File: src/test/java/cz/cvut/fel/plichjan/distmesh/FixMeshTest.java
package cz.cvut.fel.plichjan.distmesh;

import cz.cvut.fel.plichjan.distmesh.matlab.Matlab;
import cz.cvut.fel.plichjan.distmesh.result.FixedMesh;
import delaunay.Pnt;
import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

public class FixMeshTest {
    @Test
    public void testCallPrimitives() throws Exception {
        // triangle
        final FixedMesh call = new FixMesh().call(
                new double[][]{{0, 0}, {1, 0}, {0, 1}},
                new int[][]{{0, 1, 2}, {0, 2, 1}}
        );
        Assert.assertArrayEquals("Compare triangle i = " + 0, new int[]{0, 1, 2}, call.getT()[0]);
        Assert.assertArrayEquals("Compare triangle i = " + 1, new int[]{2, 0, 1}, call.getT()[1]);
    }

    @Test
    public void testCall() throws Exception {
        final List<Pnt> p = Matlab.asPntList(new double[][]{{0, 0}, {1, 0}, {0, 1}});
        final List<int[]> t = Arrays.asList(new int[][]{{0, 1, 2}, {0, 2, 1}});

        final FixedMesh call = new FixMesh().call(p, t);
        Assert.assertArrayEquals("Compare triangle i = " + 0, new int[]{0, 1, 2}, call.getT()[0]);
        Assert.assertArrayEquals("Compare triangle i = " + 1, new int[]{2, 0, 1}, call.getT()[1]);
    }

    @Test
    public void testFixElementOrientation() throws Exception {
        final List<Pnt> p = Matlab.asPntList(new double[][]{{0, 0}, {1, 0}, {0, 1}});
        final List<int[]> t = Arrays.asList(new int[][]{{0, 1, 2}, {0, 2, 1}});

        new FixMesh().fixElementOrientation(p, t);
        Assert.assertArrayEquals("Compare triangle i = " + 0, new int[]{0, 1, 2}, t.get(0));
        Assert.assertArrayEquals("Compare triangle i = " + 1, new int[]{2, 0, 1}, t.get(1));
    }
}

File: src/test/kotlin/cz/cvut/fel/plichjan/distmesh/FixMeshTest.kt
package cz.cvut.fel.plichjan.distmesh

import cz.cvut.fel.plichjan.distmesh.matlab.Matlab
import delaunay.Pnt
import org.junit.Assert.assertArrayEquals
import org.junit.Test
import java.util.*

class FixMeshTest {
    @Test
    @Throws(Exception::class)
    fun testCallPrimitives() {
        // triangle
        val call = FixMesh().call(
            arrayOf(doubleArrayOf(0.0, 0.0), doubleArrayOf(1.0, 0.0), doubleArrayOf(0.0, 1.0)),
            arrayOf(intArrayOf(0, 1, 2), intArrayOf(0, 2, 1))
        )
        assertArrayEquals("Compare triangle i = 0", intArrayOf(0, 1, 2), call.t!![0])
        assertArrayEquals("Compare triangle i = 1", intArrayOf(2, 0, 1), call.t!![1])
    }

    @Test
    @Throws(Exception::class)
    fun testCall() {
        val p = Matlab.asPntList(arrayOf(doubleArrayOf(0.0, 0.0), doubleArrayOf(1.0, 0.0), doubleArrayOf(0.0, 1.0)))
        val t = listOf(intArrayOf(0, 1, 2), intArrayOf(0, 2, 1))

        val call = FixMesh().call(p, t)
        assertArrayEquals("Compare triangle i = 0", intArrayOf(0, 1, 2), call.t!![0])
        assertArrayEquals("Compare triangle i = 1", intArrayOf(2, 0, 1), call.t!![1])
    }

    @Test
    @Throws(Exception::class)
    fun testFixElementOrientation() {
        val p = Matlab.asPntList(arrayOf(doubleArrayOf(0.0, 0.0), doubleArrayOf(1.0, 0.0), doubleArrayOf(0.0, 1.0)))
        val t = listOf(intArrayOf(0, 1, 2), intArrayOf(0, 2, 1))

        FixMesh().fixElementOrientation(p, t)
        assertArrayEquals("Compare triangle i = 0", intArrayOf(0, 1, 2), t[0])
        assertArrayEquals("Compare triangle i = 1", intArrayOf(2, 0, 1), t[1])
    }
}
